modules/alloy/src/test/ts/browser/ui/form/HtmlSelectTest.ts

```diff
@@ -1,52 +1,91 @@
-import { ApproxStructure, Assertions, Logger, Pipeline, Step } from '@ephox/agar';
-import { UnitTest } from '@ephox/bedrock-client';
+import { ApproxStructure, Assertions } from '@ephox/agar';
+import { Assert, context, describe, it } from '@ephox/bedrock-client';
 
+import { Representing } from 'ephox/alloy/api/behaviour/Representing';
+import { AlloyComponent } from 'ephox/alloy/api/component/ComponentApi';
 import * as GuiFactory from 'ephox/alloy/api/component/GuiFactory';
 import * as GuiSetup from 'ephox/alloy/api/testhelpers/GuiSetup';
 import { HtmlSelect } from 'ephox/alloy/api/ui/HtmlSelect';
-import * as RepresentPipes from 'ephox/alloy/test/behaviour/RepresentPipes';
 
-UnitTest.asynctest('HtmlSelectTest', (success, failure) => {
+describe('HtmlSelectTest', () => {
+  const simpleAssert = (label: string, expected: any, component: AlloyComponent) =>
+    Assert.eq(label, expected, Representing.getValue(component));
 
-  const sNoInitialValue = Step.async((next, die) => {
-    GuiSetup.setup((_store, _doc, _body) => GuiFactory.build(
-      HtmlSelect.sketch({
-        dom: { }, // is always a select
+  context('no initial value', () => {
+    const store = GuiSetup.bddSetup(
+      () => GuiFactory.build(
+        HtmlSelect.sketch({
+          dom: {}, // is always a select
 
-        options: [
-          { value: 'alpha', text: 'Alpha' },
-          { value: 'beta', text: 'Beta' },
-          { value: 'gamma', text: 'Gamma' }
-        ]
-      })
-    ), (_doc, _body, _gui, component, _store) => [
-      RepresentPipes.sAssertValue('Checking initial value', 'alpha', component),
-      RepresentPipes.sSetValue(component, 'beta'),
-      RepresentPipes.sAssertValue('Checking value after valid set', 'beta', component),
-      RepresentPipes.sSetValue(component, 'delta'),
-      RepresentPipes.sAssertValue('Checking value after invalid set (should still be on beta)', 'beta', component),
-      RepresentPipes.sSetValue(component, 'gamma'),
-      RepresentPipes.sAssertValue('Checking value after valid set (should now be gamma)', 'gamma', component)
-    ], next, die);
+          options: [
+            { value: 'alpha', text: 'Alpha' },
+            { value: 'beta', text: 'Beta' },
+            { value: 'gamma', text: 'Gamma' }
+          ]
+        })
+      )
+    );
+
+    it('TBA: Run tests', () => {
```

Comment:
I'm guessing this tests the representing behavior:

Maybe something like:
```suggestion
    it('should set/get the selected value using the Representing', () => {
```